ios/Plugin/Plugin.swift

```diff
@@ -20,51 +20,47 @@ public class FirebaseAnalytics: CAPPlugin {
     /// @web only. 
     /// This is a helper method to provide common APIs accross platforms.
     @objc func initializeFirebase(_ call: CAPPluginCall) {
-        call.success()
+        call.resolve()
     }
     
     /// Sets the user ID property.
     /// - Parameter call: userId - unique identifier of the user to log
     @objc func setUserId(_ call: CAPPluginCall) {
-        if !call.hasOption("userId") {
-            call.error("userId property is missing")
-            return
+        guard let userId = call.getString("userId") else {
+             call.reject("userId property is missing")
+             return
         }
         
-        let userId = call.getString("userId")
         Analytics.setUserID(userId)
-        call.success()
+        call.resolve()
     }
     
     
     /// Sets a user property to a given value.
     /// - Parameter call: name - The name of the user property to set.
     ///                   value - The value of the user property.
     @objc func setUserProperty(_ call: CAPPluginCall) {
-        if !call.hasOption("name") {
-            call.error("name property is missing")
+        guard let name = call.getString("name") else {
+            call.reject("name property is missing")
             return
         }
         
-        if !call.hasOption("value") {
-            call.error("value property is missing")
+        guard let value = call.getString("value") else {
+            call.reject("value property is missing")
             return
         }
         
-        let name = call.getString("name")
-        let value = call.getString("value")
-        
-        Analytics.setUserProperty(value, forName: name!)
-        call.success()
+        Analytics.setUserProperty(value, forName: name)
+        call.resolve()
     }
     
     
     /// Retrieves the app instance id from the service.
     /// - Parameter call: instanceId - current instance if of the app
     @objc func getAppInstanceId(_ call: CAPPluginCall) {
         let instanceId = Analytics.appInstanceID()
-        call.success([
-            "instanceId": instanceId
+        call.resolve([
+            "instanceId": instanceId ?? "0"
```

Comment:
This is not in line with the previous behaviour, do you think rejecting the call if `instanceId` is `nil` is appropriate? I've only made this change because there's a warning here as we need to unwrap `instanceId` from `Analytics.appInstanceID()` or provide a default value

Reply:
The Android implementation currently throws an error if the app instance ID is unable to be retrieved so I'll match that behaviour for now unless we want something else